.github/actions/install-dashboards/action.yml

```diff
@@ -68,14 +80,30 @@ runs:
         node-version: ${{ steps.tool-versions.outputs.node_version }}
         registry-url: 'https://registry.npmjs.org'
 
-    - run: |
+    - name: Setup Opensearch Dashboards on Linux
+      if: ${{ runner.os == 'Linux' }}
+      run: |
         npm uninstall -g yarn
         echo "Installing yarn ${{ steps.tool-versions.outputs.yarn_version }}"
         npm i -g yarn@${{ steps.tool-versions.outputs.yarn_version }}
+        yarn cache clean
+        yarn add sha.js
       working-directory: OpenSearch-Dashboards
       shell: bash
 
-    - uses: nick-fields/retry@v2
+    - name: Setup Opensearch Dashboards on Windows
+      if: ${{ runner.os == 'Windows' }}
+      run: |
+        npm uninstall -g yarn
+        echo "Installing yarn ${{ steps.tool-versions.outputs.yarn_version }}"
+        npm i -g yarn@${{ steps.tool-versions.outputs.yarn_version }}
+        yarn cache clean
+        yarn add sha.js
+      working-directory: OpenSearch-Dashboards
+      shell: pwsh
```

Comment:
this looks identical to the step above except shell type. you can use action conditions to specify the shell at run time